src/es/components/molecules/checkoutColorStage/CheckoutColorStage.js

```diff
@@ -0,0 +1,112 @@
+// @ts-check
+import { Shadow } from '../../web-components-toolbox/src/es/components/prototypes/Shadow.js'
+
+/**
+* @export
+* @class Contact
+* @type {CustomElementConstructor}
+*/
+export default class Contact extends Shadow() {
+  constructor (options = {}, ...args) {
+    super({ importMetaUrl: import.meta.url, mode: 'false', ...options }, ...args)
+
+    this.backLinkListener = this.backLinkListener.bind(this)
+  }
+
+  connectedCallback () {
+    if (this.shouldRenderCSS()) this.renderCSS()
+    if (this.shouldRenderHTML()) {
+      this.renderHTML().then(() => {
+        if (!this.backButton) {
+          this.backButton = this.root.querySelector('o-grid').shadowRoot.querySelector('.back-button')
+          this.backButton.addEventListener('click', this.backLinkListener)
+        }
+      })
+    }
+  }
+
+  disconnectedCallback () {
+    this.backButton.removeEvenetListener('click', this.backLinkListener)
```

Comment:
It would be preferable to add and remove the event listeners in sync. Async could have bad side effects, in case the component would quickly connect and disconnect before the async function got triggered. In any case, you could leave it the way it is, but I would check:

`if (this.backButton) this.backButton.removeEvenetListener('click', this.backLinkListener)`

at disconnect. To avoid direct errors.